File: Practice/AOJ/ITP1/ITP1_7B/itp1_7b.rb
loop do
    ans = 0
    n, x = gets.chomp.split.map(&:to_i)
    break if (n+x).zero?
    (1..n).each do |a|
        (a+1..[n, x-a].min).each do |b|
            (b+1..[n, x-b].min).each do |c|
                ans += a+b+c==x ? 1 : 0
            end
        end
    end
    puts ans
end

File: Practice/AOJ/ITP1/ITP1_7B/itp1_7b.py
while True:
    s = 0
    n, x = map(int, input().split())
    if n == 0 and x == 0:
        break
    for a in range(1, n+1):
        for b in range(a+1, min(n+1, x-a)):
            for c in range(b+1, min(n+1, x-b)):
                s += 1 if a+b+c == x else 0
    print(s)
